### PR TITLE
Integration tests

### LINKED ISSUE #77: Port integration tests to deis/workflow
The existing deis/deis integration tests still largely apply to the Deis v2 workflow, and they should constitute the acceptance criteria for the alpha release.

### PR DESCRIPTION
**Pull request by mboersma**
Porting the deis/deis [integration tests](https://github.com/deis/deis/tree/master/tests) to Deis v2 and [ginkgo](http://onsi.github.io/ginkgo/). 

I'm putting this out here early for feedback. My goal is to add the app tests, then get this merged so others can help flesh out the remaining tests.

Closes #77.

### CODE REVIEW

**Comment by arschles** [comment]
@mboersma would you consider putting these tests into a new repo after we've released alpha? it seems to span more than just workflow (doing git pushes, etc...)

**Comment by mboersma** [comment]
> putting these tests into a new repo...

Sure, this just seemed like the best fit for them currently since they depend on the `deis` client and (mostly) exercise the Deis v2 API. If deis/workflow ends up as deis/deis in the end, then that's probably a reasonable place for the tests to stay. Otherwise maybe a deis/integration-tests repo?

**Comment by arschles** [comment]
I'm :+1: on `deis/integration-tests`. no rush like I said. something for next year maybe

**Comment by arschles** [comment]
also, :100: on using the ginkgo/gomega suite here

File: _tests/tests_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/config"
+
```
Comment (bacongobbler) [inline _tests/tests_suite_test.go]: extra whitespace

File: _tests/tests_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	. "github.com/onsi/gomega"
+
+	"testing"
```
Comment (bacongobbler) [inline _tests/tests_suite_test.go]: the testing package seems like it should go above the third party lib imports

**Comment by bacongobbler** [comment]
Looks like an awesome first pass! +1

File: _tests/tests_suite_test.go (hunk 2 above)
Comment (mboersma) [inline _tests/tests_suite_test.go]: I agree, but apparently that's how [goimports](https://github.com/bradfitz/goimports) orders them. :sheep:

**Comment by mboersma** [comment]
This is ready for review. There are still some specs that are stubbed-out, and a number of `// TODO:` comments to be addressed. If this passes muster, I would like to merge it and then finish refining the integration tests, with help from others if anyone is inclined.

**Pull request merged**